lib/engine/step/special_buy_train.rb

```diff
@@ -22,7 +22,7 @@ def blocks?
 
       def process_buy_train(action)
         company = action.entity
-        corporation = company.owner
+        corporation = @game.current_entity
```

Comment:
I prefer to use @round.current_operator because current_entity can also be a player in some circumstances.